mdata/cache/ccache.go

```diff
@@ -157,6 +157,38 @@ func (c *CCache) Add(metric schema.AMKey, prev uint32, itergen chunk.IterGen) {
 	c.accnt.AddChunk(metric, itergen.Ts, itergen.Size())
 }
 
+func (c *CCache) AddRange(metric schema.AMKey, prev uint32, itergens []chunk.IterGen) {
+	if len(itergens) == 0 {
+		return
+	}
+	c.Lock()
+	defer c.Unlock()
+
+	ccm, ok := c.metricCache[metric]
+	if !ok {
+		ccm = NewCCacheMetric(metric.MKey)
+		ccm.AddRange(prev, itergens)
+		c.metricCache[metric] = ccm
+
+		// if we do not have this raw key yet, create the entry with the association
+		ccms, ok := c.metricRawKeys[metric.MKey]
+		if !ok {
+			c.metricRawKeys[metric.MKey] = map[schema.Archive]struct{}{
+				metric.Archive: {},
+			}
+		} else {
+			// otherwise, make sure the association exists
+			ccms[metric.Archive] = struct{}{}
+		}
+	} else {
+		ccm.AddRange(prev, itergens)
+	}
+
+	for _, itergen := range itergens {
+		c.accnt.AddChunk(metric, itergen.Ts, itergen.Size())
```

Comment:
@Dieterbe when you looked at the cpu profiles, did this method ever show up? If so, we could batch-add those as well

Reply:
i was wondering the same thing. it would definitely help with `Failed to submit event to accounting, channel was blocked`. this is a different issue, but we it shouldn't be possible to hit that situation in normal scenarios, so from that perspective we probably want to batch-add, and also batch other things that could overload the channel (like deletes perhaps?)